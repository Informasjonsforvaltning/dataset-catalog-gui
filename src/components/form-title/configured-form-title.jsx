import _ from 'lodash';
import { reduxForm } from 'redux-form';

import FormTitle from './form-title.component';
import { schema } from './form-title.validations';
import { asyncValidateDatasetInvokePatch } from '../../entrypoints/main/router/datasets/pages/dataset-page/formsLib/asyncValidateDatasetInvokePatch';
import { yupValidation } from '../../lib/yupValidation';

const config = {
  form: 'title',
  validate: values => yupValidation(schema, values),
  shouldAsyncValidate: _.stubTrue, // override default, save even if sync validation fails
  asyncValidate: asyncValidateDatasetInvokePatch
};

export const ConfiguredFormTitle = reduxForm(config)(FormTitle);
