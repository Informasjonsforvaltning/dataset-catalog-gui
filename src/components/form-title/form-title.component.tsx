import React, { memo, FC } from 'react';
import { compose } from 'redux';
import { Field, FieldArray } from 'redux-form';

import {
  withTranslations,
  Props as TranslationsProps
} from '../../providers/translations';

import Helptext from '../helptext/helptext.component';
import InputField from '../fields/field-input/field-input.component';
import LinkReadonlyField from '../fields/field-link-readonly/field-link-readonly.component';
import InputFieldReadonly from '../fields/field-input-readonly/field-input-readonly.component';
import MultilingualField from '../multilingual-field/multilingual-field.component';
import TextAreaField from '../fields/field-textarea/field-textarea.component';

interface ExternalProps {
  languages: any[];
  isReadOnly: boolean;
}

interface Props extends ExternalProps, TranslationsProps {}

const FormTitle: FC<Props> = ({
  languages,
  isReadOnly,
  translationsService
}) => (
  <form>
    <div className='form-group'>
      <Helptext
        title={translationsService.translate('schema.title.helptext.title')}
        required
        term='Dataset_title'
      />
      <MultilingualField
        name='title'
        component={isReadOnly ? InputFieldReadonly : InputField}
        languages={languages}
        label={translationsService.translate('schema.title.titleLabel')}
      />
    </div>
    <div className='form-group'>
      <Helptext
        title={translationsService.translate(
          'schema.title.helptext.description'
        )}
        required
        term='Dataset_description'
      />
      <MultilingualField
        name='description'
        component={isReadOnly ? InputFieldReadonly : TextAreaField}
        languages={languages}
        label={translationsService.translate('schema.title.descriptionLabel')}
      />
    </div>

    <div className='form-group'>
      <Helptext
        title={translationsService.translate(
          'schema.title.helptext.landingPage'
        )}
        term='Dataset_landingpage'
      />
      <FieldArray
        name='landingPage'
        isReadOnly={isReadOnly}
        component={({ fields }: any) => (
          <div>
            {fields.map((item: any, index: number) => (
              <Field
                key={index}
                name={`${item}`}
                component={isReadOnly ? LinkReadonlyField : InputField}
                label='Landingsside'
              />
            ))}
          </div>
        )}
      />
    </div>
  </form>
);

export default compose<FC<ExternalProps>>(memo, withTranslations)(FormTitle);
