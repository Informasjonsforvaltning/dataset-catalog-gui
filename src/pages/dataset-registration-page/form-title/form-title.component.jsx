import React from 'react';
import PropTypes from 'prop-types';
import { Field, FieldArray } from 'redux-form';

import localization from '../../../services/localization';
import { Helptext } from '../../../components/helptext/helptext.component';
import InputField from '../../../components/fields/field-input/field-input.component';
import LinkReadonlyField from '../../../components/fields/field-link-readonly/field-link-readonly.component';
import InputFieldReadonly from '../../../components/fields/field-input-readonly/field-input-readonly.component';
import MultilingualField from '../../../components/multilingual-field/multilingual-field.component';
import TextAreaField from '../../../components/fields/field-textarea/field-textarea.component';

export const renderLandingpage = componentProps => (
  <div>
    {componentProps.fields.map((item, index) => (
      <Field
        key={index}
        name={`${item}`}
        component={componentProps.isReadOnly ? LinkReadonlyField : InputField}
        label="Landingsside"
      />
    ))}
  </div>
);

export const FormTitle = ({ languages, isReadOnly }) => (
  <form>
    <div className="form-group">
      <Helptext
        title={localization.schema.title.helptext.title}
        required
        term="Dataset_title"
      />
      <MultilingualField
        name="title"
        component={isReadOnly ? InputFieldReadonly : InputField}
        languages={languages}
        label={localization.schema.title.titleLabel}
      />
    </div>
    <div className="form-group">
      <Helptext
        title={localization.schema.title.helptext.description}
        required
        term="Dataset_description"
      />
      <MultilingualField
        name="description"
        component={isReadOnly ? InputFieldReadonly : TextAreaField}
        languages={languages}
        label={localization.schema.title.descriptionLabel}
      />
    </div>
    <div className="form-group">
      <Helptext
        title={localization.schema.title.helptext.objective}
        term="Dataset_objective"
        required
      />
      <MultilingualField
        name="objective"
        component={isReadOnly ? InputFieldReadonly : TextAreaField}
        label={localization.schema.title.objectiveLabel}
        languages={languages}
      />
    </div>

    <div className="form-group">
      <Helptext
        title={localization.schema.title.helptext.landingPage}
        term="Dataset_landingpage"
      />
      <FieldArray
        name="landingPage"
        isReadOnly={isReadOnly}
        component={renderLandingpage}
      />
    </div>
  </form>
);

FormTitle.defaultProps = {
  languages: [],
  isReadOnly: false
};

FormTitle.propTypes = {
  languages: PropTypes.array,
  isReadOnly: PropTypes.bool
};
