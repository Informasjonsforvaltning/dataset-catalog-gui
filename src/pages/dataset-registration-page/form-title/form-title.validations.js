/* eslint-disable func-names */

import * as Yup from 'yup';

import localization from '../../../services/localization';

export const schema = Yup.object().shape({
  title: Yup.object().shape({
    nb: Yup.string()
      .nullable()
      .test(function() {
        const { nb, nn, en } = this.parent;
        if (!nb && !nn && !en) {
          return this.createError({
            message: localization.validation.required,
            path: this.path
          });
        }
        return true;
      }),
    nn: Yup.string()
      .nullable()
      .test(function() {
        const { nb, nn, en } = this.parent;
        if (!nb && !nn && !en) {
          return this.createError({
            message: localization.validation.required,
            path: this.path
          });
        }
        return true;
      }),
    en: Yup.string()
      .nullable()
      .test(function() {
        const { nb, nn, en } = this.parent;
        if (!nb && !nn && !en) {
          return this.createError({
            message: localization.validation.required,
            path: this.path
          });
        }
        return true;
      })
  }),
  description: Yup.object().shape({
    nb: Yup.string()
      .nullable()
      .test(function() {
        const { nb, nn, en } = this.parent;
        if (!nb && !nn && !en) {
          return this.createError({
            message: localization.validation.required,
            path: this.path
          });
        }
        return true;
      }),
    nn: Yup.string()
      .nullable()
      .test(function() {
        const { nb, nn, en } = this.parent;
        if (!nb && !nn && !en) {
          return this.createError({
            message: localization.validation.required,
            path: this.path
          });
        }
        return true;
      }),
    en: Yup.string()
      .nullable()
      .test(function() {
        const { nb, nn, en } = this.parent;
        if (!nb && !nn && !en) {
          return this.createError({
            message: localization.validation.required,
            path: this.path
          });
        }
        return true;
      })
  }),
  landingPage: Yup.array().of(
    Yup.string().url(localization.validation.validateLink)
  )
});

/* eslint-enable func-names */
