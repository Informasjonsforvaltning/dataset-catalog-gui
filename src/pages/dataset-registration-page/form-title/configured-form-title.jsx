import _ from 'lodash';
import { reduxForm } from 'redux-form';

import { FormTitle } from './form-title.component';
import { schema } from './form-title.validations';
import { asyncValidateDatasetInvokePatch } from '../formsLib/asyncValidateDatasetInvokePatch';
import { yupValidation } from '../../../lib/yupValidation';

const config = {
  form: 'title',
  validate: values => yupValidation(schema, values),
  shouldAsyncValidate: _.stubTrue, // override default, save even if sync validation fails
  asyncValidate: asyncValidateDatasetInvokePatch
};

export const ConfiguredFormTitle = reduxForm(config)(FormTitle);
