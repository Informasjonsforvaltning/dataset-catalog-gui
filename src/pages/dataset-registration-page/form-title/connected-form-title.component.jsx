import { getFormSyncErrors } from 'redux-form';
import { connect } from 'react-redux';
import _ from 'lodash';

import { emptyArray } from '../../../schemaTypes';
import { ConfiguredFormTitle } from './configured-form-title';

const mapStateToProps = (state, { datasetItem = {} }) => ({
  initialValues: {
    title: datasetItem.title || {},
    description: datasetItem.description || {},
    objective: datasetItem.objective || {},
    landingPage: _.get(datasetItem, 'landingPage', emptyArray)
  },
  syncErrors: getFormSyncErrors('title')(state)
});

export const ConnectedFormTitle = connect(mapStateToProps)(ConfiguredFormTitle);
