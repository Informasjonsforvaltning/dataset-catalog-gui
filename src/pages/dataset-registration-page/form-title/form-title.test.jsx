import React from 'react';
import { shallow } from 'enzyme';
import { FormTitle } from './form-title.component';

let defaultProps;
let wrapper;

beforeEach(() => {
  defaultProps = {};
  wrapper = shallow(<FormTitle {...defaultProps} />);
});

test('should render FormTitle, { renderLandingpage  correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
