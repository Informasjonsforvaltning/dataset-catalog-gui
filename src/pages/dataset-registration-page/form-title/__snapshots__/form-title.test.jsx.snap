// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FormTitle, { renderLandingpage  correctly 1`] = `
<form>
  <div
    className="form-group"
  >
    <withStateHandlers(Component)
      required={true}
      term="Dataset_title"
      title="Tittel"
    />
    <MultilingualField
      component={[Function]}
      label="Tittel"
      languages={Array []}
      name="title"
      showLabel={false}
    />
  </div>
  <div
    className="form-group"
  >
    <withStateHandlers(Component)
      required={true}
      term="Dataset_description"
      title="Beskrivelse av datasettet"
    />
    <MultilingualField
      component={[Function]}
      label="Beskrivelse"
      languages={Array []}
      name="description"
      showLabel={false}
    />
  </div>
  <div
    className="form-group"
  >
    <withStateHandlers(Component)
      required={true}
      term="Dataset_objective"
      title="Formålet med datasettet"
    />
    <MultilingualField
      component={[Function]}
      label="Formål"
      languages={Array []}
      name="objective"
      showLabel={false}
    />
  </div>
  <div
    className="form-group"
  >
    <withStateHandlers(Component)
      term="Dataset_landingpage"
      title="Lenke til mer informasjon om datasettet"
    />
    <FieldArray
      component={[Function]}
      isReadOnly={false}
      name="landingPage"
    />
  </div>
</form>
`;
